### PR TITLE
ASC-378 impliment mark validation

### PR DESCRIPTION
**Pull request by zreichert**
- configuration now accepts value_match as a param to given pytest_mark
- value 'uuid' will validate that the passed string is a uuid example: value_match=uuid
- any other value will be treated as a regex and a match will be attempted, white space is not permitted

There is also a testing improvement to the contest.py file.

### CODE REVIEW

File: pytest_mark_checker.py [hunk 1]
```diff
@@ -84,3 +85,46 @@ def rule_M5XX(self, node, rule_name, rule_conf):
...
+            if re.search(r'^test_', node.name):
+                for decorator in node.decorator_list:
+                    try:
+                        value = decorator.args[0].s
+                        mark_key = decorator.func.attr
+                        decorator_type = decorator.func.value.value.id
+                        # must be marked and configured to match in order to match the content :)
+                        if all([decorator_type == 'pytest', mark_key == rule_conf['name'], rule_conf['value_match']]):
+                            configured = True
+                            if rule_conf['value_match'] == 'uuid':
```
Comment (ryan-rs) [inline pytest_mark_checker.py]: I would suggest not overloading the "value_match" config option to take both static matchers and custom user defined regex matchers. Instead I propose you separate the config options:

```
[flake8]
pytest_mark1 = name=test_id
                         value_regex=^this_is_a_regex
pytest_mark2 = name=test_other_id
                         value_match=uuid
```

File: tests/conftest.py [hunk 2]
```diff
@@ -10,3 +13,9 @@ def assert_lines(expected, observed):
     e.sort()
```
Comment (ryan-rs) [inline tests/conftest.py]: Add a docstring to this function to explain it's function.

File: pytest_mark_checker.py (hunk 1 above)
Comment (zreichert) [reply pytest_mark_checker.py]: Do you have an opinion on which one takes precedence?

Comment (ryan-rs) [reply pytest_mark_checker.py]: @zreichert If a user provided a regex I would give that one precedence.

**Pull request merged**